plugin.rb

```diff
@@ -146,14 +146,21 @@ def index
     end
   end
 
+  add_to_class(:user, :can_use_canned_replies?) do
+    return true if staff?
+    group_list = SiteSetting.canned_replies_groups.split("|").map!(&:downcase)
```

Comment:
We should be careful when using "bang" methods. In this case, it's harmless, since the "`split("|")`" is creating a new array object, but if you were to use it directly on the site setting object, it would modify it.

```suggestion
    group_list = SiteSetting.canned_replies_groups.split("|").map(&:downcase)
```